packages/types/src/dal/repository-service.ts

```diff
@@ -70,6 +71,12 @@ export interface RepositoryService<T = any> extends BaseRepositoryService<T> {
   ): Promise<[T[], Record<string, unknown[]>]>
 
   upsert(data: any[], context?: Context): Promise<T[]>
+
+  upsertWithReplace(
+    data: any[],
+    upsertConfig?: UpsertWithReplaceConfig<T>,
```

Comment:
suggestion: Would recommend scoping the variable names with `upsertWithReplace` as well since its not doing just an upsert or change to config if there is a context of the function

```suggestion
    config?: UpsertWithReplaceConfig<T>,
```